mysql/mysqldump.go

```diff
@@ -0,0 +1,578 @@
+			column.Ignored.Default = true
+		case ast.ColumnOptionUniqKey:
+			column.Unique = true
+		case ast.ColumnOptionCheck:
+			column.Ignored.Check = true
+		case ast.ColumnOptionReference:
+			column.Ignored.ForeignKey = true
+		default:
+			conv.Unexpected(fmt.Sprintf("Not supported type option of const value `%v` found in table : %s , column : %s. Mapping of columnOptionType can be found here: 'https://godoc.org/github.com/pingcap/parser/ast#ColumnOptionType'", op, tableName, column.Name))
+		}
+	}
+	return isPk
+}
+
+//getTypeModsAndID returns ID and mods of column datatype
+func getTypeModsAndID(columnType string) (string, []int64) {
+	//There are no methods in pincap parser to retirieve ID and mods
+	//We will process columnType eg:'varchar(40)' and split ID from the string.
+	//We retrieve mods using regex expression and convert it to INT64
+	id := columnType
+	var mods []int64
+	if strings.Contains(columnType, "(") {
+		id = strings.Split(columnType, "(")[0]
+		// For 'set' and 'enum' datatypes, values provided are not maxLength
+		if id == "set" || id == "enum" {
+			return id, nil
+		}
+		r, _ := regexp.Compile("\\((.*?)\\)")
+		values := r.FindString(columnType)
+		strMods := strings.Split(values[1:len(values)-1], ",")
+		for _, i := range strMods {
+			j, err := strconv.ParseInt(i, 10, 64)
+			if err != nil {
+				panic(err)
+			}
+			mods = append(mods, j)
+		}
+	}
+	// 'BINARY' keyword will be added to all blob datatypes by parser.
+	// Eg: mediumblob BINARY. It needs to be ignored to retrieve ID
+	if strings.Contains(id, " ") {
+		id = strings.Split(columnType, " ")[0]
+	}
+	return id, mods
+}
+
+// Error can be due to unsupported Spatial datatypes in Create statement or some invalid entries
+// in insert statement. handleParseError attempts at creating parsable query.
+func handleParseError(conv *internal.Conv, query string, err error, l [][]byte) ([]ast.StmtNode, bool) {
+	new_query := query
+	// Check if error is due to Insert statement.
+	r, _ := regexp.Compile("INSERT\\sINTO\\s(.*?)\\sVALUES\\s")
+	insert_stmt := r.FindString(new_query)
+	if insert_stmt != "" {
+		// Likely causes of failing to parse Insert statement:
+		// a) Due to some invalid value
+		// b) Query size is more than what pingcap parser could handle (more than 40MB in size)
+		return handleInsertStatement(conv, query, insert_stmt)
+	}
+
+	// Check if error is due to spatial datatype as it is not supported by Pingcap parser.
+	var isSpatial bool
+	for _, spatial := range internal.Spatial {
+		if strings.Contains(strings.ToLower(err.Error()), spatial) && isSpatial == false {
+			isSpatial = true
+			conv.Unexpected(fmt.Sprintf("Unsupported datatype '%s' encountered while parsing following statement at line number %d : \n%s", spatial, len(l), query))
+			internal.VerbosePrintf("Converting datatype '%s' to 'Text' and retrying to parse the statement\n", spatial)
+		}
+	}
+	if isSpatial {
+		return handleSpatialDatatype(conv, query, l)
+	}
+
+	return nil, true
+}
```

Comment:
> Where is `Spatial` used? Are these types related to `mysql`? I am not sure if this is a typical way how Go organizes configurations, do you have any reference?

@hengfengli We use `spatial` to handle the spatial datatypes of mysql. It will be used for Information schema approach also. We can declare `spatial` array in function also, but thought of creating a config.go - which can be used by other packages also, and we can also add some common variables for future enhancements. 

What are your views on this?